src/group/managed_group/events.rs

```diff
@@ -0,0 +1,268 @@
+use crate::credentials::*;
+use crate::group::*;
+use crate::messages::proposals::*;
+use crate::schedule::psk::*;
+
+/// Group event that occured while processing messages in `ManagedGroup`.
+#[derive(Debug, PartialEq, Clone)]
+pub enum GroupEvent {
+    MemberAdded(MemberAddedEvent),
+    MemberRemoved(MemberRemovedEvent),
+    MemberUpdated(MemberUpdatedEvent),
+    PskReceived(PskReceivedEvent),
+    ReInit(ReInitEvent),
+    ApplicationMessage(ApplicationMessageEvent),
+    InvalidMessage(InvalidMessageEvent),
+    Error(ErrorEvent),
+}
+
+// Member events
+
+/// Event that occurs when member `sender` adds member `added_member`.
+#[derive(Debug, PartialEq, Clone)]
+pub struct MemberAddedEvent {
+    aad: Vec<u8>,
+    sender: Credential,
+    added_member: Credential,
+}
+
+impl MemberAddedEvent {
+    pub(crate) fn new(aad: Vec<u8>, sender: Credential, added_member: Credential) -> Self {
+        Self {
+            aad,
+            sender,
+            added_member,
+        }
+    }
+
+    /// Get a reference to the member added event's aad.
+    pub fn aad(&self) -> &[u8] {
+        &self.aad
+    }
+
+    /// Get a reference to the member added event's sender.
+    pub fn sender(&self) -> &Credential {
+        &self.sender
+    }
+
+    /// Get a reference to the member added event's added member.
```

Comment:
Not sure I understand these comments ... 😄 Maybe simplify the sentences for all these comments.